tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -518,3 +524,450 @@ def test_io_data_server_pod_restart(self):
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("STEP 4: Performed WRITEs/READs/Verify with variable sizes objects in "
+                    "degraded mode")
+
+        LOGGER.info("STEP 5: Restore data pod and check cluster status.")
+        pod = const.POD_NAME_PREFIX
+        self.restore_method = self.pod_dict.get(pod)[-1]
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={
+                                           "deployment_name": self.pod_dict.get(pod)[-2],
+                                           "deployment_backup": self.deployment_backup,
+                                           "num_replica": self.pod_dict.get(pod)[2],
+                                           "set_name": self.pod_dict.get(pod)[1]})
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        LOGGER.info("STEP 5: Successfully started data pod and cluster is online.")
+        self.restore_pod = False
+
+        LOGGER.info("STEP 6: Perform READ/Verify on data written with buckets created in healthy "
+                    "mode in step-1")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("STEP 6: Successfully run READ/Verify on data written with buckets created in "
+                    "healthy mode in step-1")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("STEP 7: Perform READs and verify DI on the data written on buckets "
+                        "created in degraded mode in step-4")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False, nsamples=5, nclients=5)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("STEP 7: Successfully run READ/Verify on data written on buckets created "
+                        "in degraded mode in step-4")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("STEP 8: Create new IAM user and buckets, Perform WRITEs-READs-Verify with "
```

Comment:
here as well instead of whole step under DTM flag, just IAM user creation and creation of bucket can go under DTM flag

Reply:
Also, no need of same if condition multiple times, step under this can go in above if condition as both are same.